### DISCUSSION TITLE
Golang Lsp Server workspace seems confused

### DISCUSSION THREAD

**Discussion by nvima** [VonHeikemen/lsp-zero.nvim#107] [Q&A] [answered]
Hey, i changed my neovim config some weeks ago where i switched to lsp zero. Before i had not this problem with my project.

In my go projects with this folder structure, the lsp seems confused where the root workdir is and i get errors from import inside a single golang project.

ex. i have a folder with all my go projects:
/goprojects/project1
/goprojects/project2

in ex. this file
/goprojects/project1/pkg/utils/get-id.go
i get errors where i can not import a file from "project1/pkg/types"
And i get errors when i use variables i declared in the same folder,  the lsp not recognize i declared the var inside a file in the same folder.
when i build my projects i get no error and everything works, only my lsp in neovim seems broken for this kind of goprojects where i have multiple nested go project folders.

this is my actuel lsp zero conf:
https://github.com/nvima/.dotfiles/blob/main/nvim/.config/nvim/after/plugin/lsp.lua

this is my old config where everything works fine, but i did not used lsp zero there:
https://github.com/nvima/.dotfiles/blob/main/nvim/.config/nvim_old/lua/user/lsp/configs.lua

Can someone help me configure this right?

**Comment by VonHeikemen** [comment, marked as answer]
Can you update lsp-zero and test again? I've moved some things around. I think there was a function in mason.nvim that was causing weird behaviors.

**Reply by nvima** [reply]
Yes, seems that it works now :) Thanks for your work
